### ISSUE TITLE
Input date not always populating

### ISSUE THREAD

**Issue by mrustow** [Princeton-CDH/geniza#887]
### testing notes (QA)

On the [QA admin site](https://test-geniza.cdh.princeton.edu/admin), create a new document, and click "save".

- [x] On the [QA public site search](https://test-geniza.cdh.princeton.edu/en/documents/), search for that document. In the search result, its input date should read 2023 rather than "unknown".

---

Hello party people ;) On the public site, I just sorted by input date (great feature! thanks!) earliest to latest. The earliest ones were date "unknown," and very high PGPIDs, so high that they must have been input this month. After some digging it looks like the problem began May 4 (PGPID 35277), and its latest manifestation was today (PGPID 35363) — I got those input dates from the admin site, meaning they're in the admin site somewhere, just not displaying on the public site. Correction: not displaying in the search results/list view, but they are displaying in site view. 🤷🏻‍♀️

<img width="495" alt="image" src="https://user-images.githubusercontent.com/73319225/170370732-b1e06b66-5e4a-4db8-8bb7-711727827dba.png">

**Comment by kseniaryzhova** [comment]
@blms this is still an issue, for example PGPID 36488 contains in the admin history that Alan added it Oct. 31 2022, but displays as unknown in the front end.

**Comment by rlskoeser** [comment]
This is an indexing problem. I wonder if it's possible the initial indexing happens before the log entry documenting the input date is available?

I checked the index data for this one and input date was set correctly, so I indexed it and now the date appears on the search results.

I ran a solr queries on the production Solr for records with no input date set (`NOT input_year_i:[* TO *]`) and found 14 records; I went ahead and reindexed them.

This resolves the display problem, but won't solve the underlying bug (if there is one). It would be good to see if we can duplicate it on the test site — my first thought is try creating a record but not making any changes after the first save.  If that results in this behavior, then we can work on a fix — if there are no log entries then maybe we can fallback to the `created` attribute on the document (that should work for all new records; it will only be wrong for imported records, but those _should_ all have log entries.)
